### PR TITLE
Add e2e for variable expansion

### PR DESCRIPTION
**Pull request by pmorie**
How embarrassing that nary a line of end-to-end test was composed for this feature.

@bgrant0607 @thockin @erictune

### CODE REVIEW

**Comment by pmorie** [comment]
@derekwaynecarr

**Comment by derekwaynecarr** [comment]
This looks good.

File: test/e2e/expansion.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+			ObjectMeta: api.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": podName},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:    "dapi-container",
+						Image:   "gcr.io/google_containers/busybox",
+						Command: []string{"sh", "-c", "echo \"$(TEST_VAR)\""},
```
Comment (thockin) [inline test/e2e/expansion.go]: how about Args, too?

**Comment by thockin** [comment]
Needs args, then LGTM

**Comment by pmorie** [comment]
My bad -- I dropped this in the OpenShift GA / RH Summit prep hustle.  Will complete this.

**Comment by pmorie** [comment]
@thockin @derekwaynecarr last test case added.

**Comment by zmerlynn** [comment]
LGTM

**Comment by bprashanth** [comment]
@thockin @quinton-hoole for the merge call on this

**Comment by thockin** [comment]
LGTM, thanks

**Pull request merged**